### PR TITLE
[AddressLowering] Some small fixes.

### PR DESCRIPTION
**Pull request by nate-chandler**
See individual commit messages for details.

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test

File: test/SILOptimizer/address_lowering.sil [hunk 1]
```diff
@@ -938,20 +938,20 @@ bb0(%0 : @owned $P):
 }
 
 // CHECK-LABEL: sil [ossa] @f161_testOpenedArchetype : $@convention(thin) (@in any P) -> () {
-// CHECK: bb0(%0 : $*any P):
```
Comment (atrick) [inline test/SILOptimizer/address_lowering.sil:941]: ok, but %0 is the most concise and recognizable way to refer to the first function argument.

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp [hunk 2]
```diff
@@ -3362,8 +3362,19 @@ static void rewriteFunction(AddressLoweringState &pass) {
...
+    SmallPtrSet<Operand *, 8> originalUses;
     SmallVector<Operand *, 8> uses(valueDef->getUses());
-    for (Operand *oper : uses) {
+    for (auto *oper : uses) {
+      originalUses.insert(oper);
+      UseRewriter::rewriteUse(oper, pass);
+    }
+    uses.clear();
+    for (auto *use : valueDef->getUses()) {
+      uses.push_back(use);
```
Comment (atrick) [inline lib/SILOptimizer/Mandatory/AddressLowering.cpp:3373]: I would check membership in originalUses before inserting them in a worklist.

**Review by atrick — APPROVED** [review]
A couple comments...

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp (hunk 2 above)
Comment (nate-chandler) [reply lib/SILOptimizer/Mandatory/AddressLowering.cpp:3376]: Fixed.

File: test/SILOptimizer/address_lowering.sil (hunk 1 above)
Comment (nate-chandler) [reply test/SILOptimizer/address_lowering.sil:941]: Switched this function back to the way it was for this register.

**Comment by nate-chandler** [comment]
@swift-ci please test and merge

**Comment by nate-chandler** [comment]
@swift-ci please test macOS platform

**Pull request merged**